### DISCUSSION TITLE
Create Nested Sub-menu's with Understrap

### DISCUSSION THREAD

**Discussion by rushishelat** [holger1411/understrap#1388] [Q&A] [answered]
Hi, I am wondering how to create nested sub-menu's using the understrap theme. I know how to do it in wordpress, but when i customize in wordpress it does not respond on the website. Is there a way to customize it in the files? Please let me know, thank you!

**Comment by iwasthesword** [comment]
As far as I know this is a Bootstrap thing that doesn't allow nested sub-menus. I don't think understrap should support it either but the motive should be in the docs in my opnion as this is a common question I've seen in bootstrap themes

**Comment by pattonwebz** [comment]
> As far as I know this is a Bootstrap thing that doesn't allow nested sub-menus.

This is true, BS doesn't have facilities to work with greater depths of sub-menus by choice. The walker used in this theme can be toggled to greater depths and the nav items will output in markup but BS has no JS/CSS for activating or displaying them. You would have to write that yourself or rework the nav walker markup generation.

**Comment by gabrielcastillo** [comment, marked as answer]
@rushishelat I have a new branch created.. I am currently working on a solution for having multilevel nav menu's. Hopefully, have it added to the new release.

Below is a link to my demo video

[Multi level nav menu demo](https://www.loom.com/share/b28ef7c036ab407e9201cd9dc074c720)

**Reply by LukaszJaro** [reply]
The video shows alot of the complexities of multi-level dropdown, you can see comment here why bootstrap didn't implement https://github.com/twbs/bootstrap/issues/27659#issuecomment-441783535

Make sure your solution is accessible!

However if you scroll down further https://github.com/twbs/bootstrap/issues/27659#issuecomment-675166159

It looks like the creator has reconsidered and added help wanted label to the ticket in 2020, they are now looking to restore multi-level functionality.

It would probably be better to help them on the bootstrap solution instead of trying to do custom approach for understrap only.

**Reply by howarddc** [reply]
Great idea on contributing to Bootstrap, @LukaszJaro! We just discussed doing some more work on this today, so we'll get a solid prototype going and then figure out the best options for distributing it.

**Comment by Piirka** [comment]
I have seen few megamenus used instead typical multilevel dropdown. I think those are nicer to use because it can be opened by clicking and it does not go away until you click it away. 

Just tested with CSS what would it look like and came up with this solution.

Perhaps this could be part of understrap? But I am not sure is this accessible enough.

In this example If I want to use megamenu I have set Level 1 link a class named megamenu and added w-100 class for landing page link. Other sub links are flexbox columns.

![megamenu](https://user-images.githubusercontent.com/523847/156611634-e3185a4e-a29c-4126-b2dc-8ef3ab092547.png)

![menu](https://user-images.githubusercontent.com/523847/156610572-2767ffe7-cdec-4730-a807-b6c3518b128a.png)
